test.js

```diff
@@ -69,10 +92,125 @@ test('it supports multiple arguments', t => {
   source.emit(1, 'qu');
 
   t.deepEqual(
-    sink.getReceivedData(), 
+    sink.getReceivedData(),
     ['foo','bar','baz','qu'],
     'sink gets seed and subsequent data'
   );
 
   t.end();
 });
+
+test('it queues sync completion', t => {
+  let history = [];
+
+  const seededSrc = startWith('a', 'b', 'c')(fromIter(['d']));
+
+  const makeSink = () => {
+    let talkback;
+    return (t, d) => {
+      if (t === 0) talkback = d;
+      else history.push([t,d]);
+
+      if (t !== 2) talkback(1);
+    };
+  };
+
+  seededSrc(0, makeSink());
+
+  t.deepEqual(history, [
+    [1, 'a'],
+    [1, 'b'],
+    [1, 'c'],
+    [1, 'd'],
+    [2, undefined],
+  ], 'sink gets data in the correct order');
+
+  t.end();
+});
+
+test('it doesn\'t request data when receiving uknown type', t => {
+  let history = [];
+  const report = (t,d) => t !== 0 && history.push([t,d]);
+
+  const source = makeMockCallbag(report, true);
+  const seedWithFoo = startWith('foo');
+  const seededSource = seedWithFoo(source);
+
+  const autoPull = () => {};
+  forEach(autoPull)(seededSource);
+
+  source.emit(1, 'a');
+  source.emit(1, 'b');
+  source.emit('unknown', 'c');
+  source.emit(1, 'd');
+  source.emit('unknown', 'e');
+
+  t.deepEqual(history, [
+    [1, undefined], // request sent up on 0
+    [1, undefined], // request sent up startWith seed
+    [1, undefined],
+    [1, undefined],
+    [1, undefined],
+  ], 'source gets correct number of requests from sink');
+
+  t.end();
+});
+
+test('it passes sink errors up (& data for unknown types too)', t => {
+  let history = [];
+  const report = (t,d) => t !== 0 && history.push([t,d]);
+
+  const source = makeMockCallbag(report, true);
+  const seedWithFoo = startWith('foo');
+  const sink = makeMockCallbag();
+
+  seedWithFoo(source)(0, sink);
+
+  sink.emit(1);
+  sink.emit('unknown', 'payload');
+  sink.emit(11, 'other_data');
+  sink.emit(2, 'err');
+
+  t.deepEqual(history, [
+    [1, undefined],
+    ['unknown', 'payload'],
+    [11, 'other_data'],
+    [2, 'err'],
+  ], 'source gets type & data from sink');
+
+  t.end();
+});
+
+test('it stops emitting after receiving unsubscription request', t => {
+  let history = [];
+
+  const seededSrc = startWith('a', 'b', 'c')(fromIter(['d']));
+
+  const makeSink = () => {
+    let talkback;
+    let counter = 0;
+    return (t, d) => {
+      if (t === 0) talkback = d;
+      else {
+        history.push([t,d]);
+        counter++;
+      }
+
+      if (counter === 2) {
+        talkback(2);
+        return;
+      }
+
+      if (t !== 2) talkback(1);
+    };
+  };
+
+  seededSrc(0, makeSink());
+
+  t.deepEqual(history, [
+    [1, 'a'],
+    [1, 'b'],
+  ], 'sink stops receiving data after unsubscribing');
+
+  t.end();
+});
```

Comment:
How about adding `callbag-take` as a devDependency and using that instead? Then we could just create the sink with `makeMockCallbag()`, and test against `.getReceivedData` (and maybe `.checkConnection` for good measure).